src/app/document-services/customer-item/customer-item.service.ts

```diff
@@ -1,17 +1,32 @@
 import { Injectable } from "@angular/core";
 import { CustomerItem } from "@boklisten/bl-model";
-import { ApiService } from "../../api/api.service";
 import { BL_CONFIG } from "../../bl-connect/bl-config";
-import {
-	CachedDocumentService,
-	CachedDocumentServiceOptions,
-} from "../../document/cached-document.service";
+import { CachedDocumentService } from "../../document/cached-document.service";
 import { BlDocumentService } from "../../document/bl-document.service";
+import { ApiResponse } from "../../api/api-response";
+import { ApiService } from "../../api/api.service";
 
 @Injectable()
 export class CustomerItemService extends BlDocumentService<CustomerItem> {
-	constructor(private cachedDocumentService: CachedDocumentService) {
+	constructor(
+		private cachedDocumentService: CachedDocumentService,
+		private _apiService: ApiService
+	) {
 		super(cachedDocumentService);
 		this.setCollection(BL_CONFIG.collection.customerItem);
 	}
+
+	public generateCustomerItemReport(options: {
+		branchFilter?: string[];
+		createdAfter?: string;
+		createdBefore?: string;
+		returned: boolean;
+		handout: boolean;
+		buyout: boolean;
+	}): Promise<ApiResponse> {
```

Comment:
How exactly do you intend to download the file? It doesn't download automatically when I test with `fetch` in my browser console at least. The other `bl-admin` reports use `xlsx`'s `writeFile`, which probably does sorcery. The only way I can find to download a file from a POST call is to get the result, store it in a `Blob`, call `URL.createObjectURL` and put the result in the `href` of a `<a href="..." download>` which the user must click (or be clicked with JS). This would require you to extract the file-body (not `ApiResponse`) and read the headers to find out the file name.

Reply:
Updated to do the xlsx handling on the client